newIDE/app/src/MainFrame/ElectronMainMenu.js

```diff
@@ -319,40 +178,177 @@ class ElectronMainMenu extends React.Component<MainMenuProps, {||}> {
+          },
+          { type: 'separator' },
+          {
+            label: i18n._(t`Contribute to GDevelop`),
+            onClickOpenLink: 'https://gdevelop-app.com/contribute/',
+          },
+          {
+            label: i18n._(t`Create Extensions for GDevelop`),
+            onClickOpenLink:
+              'https://github.com/4ian/GDevelop/blob/master/newIDE/README-extensions.md',
+          },
+          { type: 'separator' },
+          {
+            label: i18n._(t`Help to Translate GDevelop`),
+            onClickOpenLink: 'https://crowdin.com/project/gdevelop',
+          },
+          {
+            label: i18n._(t`Report a wrong translation`),
+            onClickOpenLink: 'https://github.com/4ian/GDevelop/issues/969',
+          },
+        ],
+      };
+      if (!isMacLike()) {
+        helpTemplate.submenu.push(
+          { type: 'separator' },
+          {
+            label: i18n._(t`About GDevelop`),
+            onClickSendEvent: 'main-menu-open-about',
+          }
+        );
+      }
+
+      const template: Array<RootMenuTemplate> = [
+        fileTemplate,
+        editTemplate,
+        viewTemplate,
+        windowTemplate,
+        helpTemplate,
       ];
-    }
 
-    if (ipcRenderer) {
-      ipcRenderer.send('set-main-menu', template);
-    }
-  }
+      if (isMacLike()) {
+        template.unshift({
+          label: i18n._(t`GDevelop 5`),
+          submenu: [
+            {
+              label: i18n._(t`About GDevelop`),
+              onClickSendEvent: 'main-menu-open-about',
+            },
+            { type: 'separator' },
+            {
+              label: i18n._(t`My Profile`),
+              onClickSendEvent: 'main-menu-open-profile',
+            },
+            {
+              label: i18n._(t`Preferences`),
+              onClickSendEvent: 'main-menu-open-preferences',
+            },
+            {
+              label: i18n._(t`Language`),
+              onClickSendEvent: 'main-menu-open-language',
+            },
+            { type: 'separator' },
+            { role: 'services', submenu: [] },
+            { type: 'separator' },
+            { role: 'hide' },
+            { role: 'hideothers' },
+            { role: 'unhide' },
+            { type: 'separator' },
+            { role: 'quit' },
+          ],
+        });
+
+        editTemplate.submenu.push(
+          { type: 'separator' },
+          {
+            label: i18n._(t`Speech`),
+            submenu: [{ role: 'startspeaking' }, { role: 'stopspeaking' }],
+          }
+        );
+
+        windowTemplate.submenu = [
+          { role: 'minimize' },
+          { role: 'zoom' },
+          { type: 'separator' },
+          { role: 'front' },
+        ];
+      }
+
+      if (ipcRenderer) {
+        ipcRenderer.send('set-main-menu', template);
+      }
+    },
+    [i18n, project]
+  );
+
+  React.useEffect(
+    () => {
+      _buildAndSendMenuTemplate();
+    },
+    [_buildAndSendMenuTemplate, i18n.language]
```

Comment:
So the effect runs if the callback changes, which is the case if i18n or project changes. Clever - almost a bit obfuscated! But it works :)

Reply:
I encountered something strange while working on this part. What I actually wanted was to keep `i18n.language` and `project` as the `_buildAndSendMenuTemplate` callback dependencies to make it a bit more clear.

But React complained about `i18n` as a missing dependency (which is fine), but if I add it then React calls `i18n.language` as an unnecessary dependency (which seems fine too). But then somewhat strangely, `i18n` does not cause the effect to rerun on language change while `i18n.language` does! So I added `i18n.language` to this effect instead, to make it work properly on language change and remove any warnings.

Reply:
Yes this is because i18n object is the same during all the lifetime of the app, while i18n.language can change - but i18n is not changing. Another way to avoid the warning that is maybe safer would be to pass language itself:
```
const language = i18n.language;
...
[_buildAndSendMenuTemplate, project, i18n, language]
```

something like this :) Up to you!

Reply:
> Another way to avoid the warning that is maybe safer would be to pass language itself

I'm already passing `i18n.language` as dependency to `useEffect` - would destructuring change anything? Or if you mean adding `language` dependency to `useCallback`, that still gives the warning. I'm not sure I'm fully clear about what you're suggesting here.

Reply:
Another option is to completely remove the intermediary `_buildAndSendMenuTemplate`, and put its content directly into a `useEffect` with dependencies as `i18n, project, language`. Then  I guess the dependency list would almost directly reflect the intention, that we want to rebuild the menu on project or language change.

Reply:
> I'm already passing i18n.language as dependency to useEffect - would destructuring change anything?

This won't change the runtime behavior, but this will avoid the linter warning, because the linter will warn you if you use `i18n, i18n.language`, but not if you use `i18n, language`.

> Another option is to completely remove the intermediary _buildAndSendMenuTemplate, and put its content directly into a useEffect with dependencies as i18n, project, language. 

That's an option, and if `_buildAndSendMenuTemplate` is not used anywhere else that might be clearer indeed.

Reply:
The useEffect does not have i18n as dependency - the useCallback has it (since the _build function uses i18n._). And adding language to the useCallback dependency list leads to unnecessary dep warning regardless of destructuring :(

> That's an option, and if `_buildAndSendMenuTemplate` is not used anywhere else that might be clearer indeed.

Yeah, the build function isn't used anywhere else in the component, so that won't be a problem. Should I push a commit for this?

Reply:
Yes that sounds fine. You can also put `buildAndSendMenuTemplate` outside of the component itself and pass to it all the arguments it requires. This allows for:
* A clean function, that has a name so is "auto documenting".
* A useEffect that is both linter compatible and very clear ("in case of change in project, i18n, language, re-run buildAndSendMenuTemplate" => make sense! We don't even need a comment).

Reply:
I've made the changes. You're right, moving the function out of the component certainly made it much better and cleaner :smile: